lib/include/currentSourceModels.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+// Standard includes
+#include <array>
+#include <functional>
+#include <string>
+#include <tuple>
+#include <vector>
+#include <cmath>
+
+// GeNN includes
+#include "codeGenUtils.h"
+// #include "neuronModels.h"
+#include "newModels.h"
+
+//----------------------------------------------------------------------------
+// Macros
+//----------------------------------------------------------------------------
+#define SET_INJECTION_CODE(INJECTION_CODE) virtual std::string getInjectionCode() const{ return INJECTION_CODE; }
+#define SET_TIME_CONDITION_CODE(TIME_CONDITION_CODE) virtual std::string getTimeConditionCode() const{ return TIME_CONDITION_CODE; }
+#define SET_EXTRA_GLOBAL_PARAMS(...) virtual StringPairVec getExtraGlobalParams() const{ return __VA_ARGS__; }
+
+//----------------------------------------------------------------------------
+// CurrentSourceModels::Base
+//----------------------------------------------------------------------------
+namespace CurrentSourceModels
+{
+//! Base class for all current source models
+class Base : public NewModels::Base
+{
+public:
+    //----------------------------------------------------------------------------
+    // Declared virtuals
+    //----------------------------------------------------------------------------
+    //! Gets the code that defines current injected each timestep 
+    virtual std::string getInjectionCode() const{ return ""; }
+
+    //! Gets code which defines the condition when the should be applied described current source model.
+    /*! This evaluates to a bool (e.g. "t > 20 && t < 40"). */
+    virtual std::string getTimeConditionCode() const{ return ""; }
+
+    //! Gets names and types (as strings) of additional parameters
+    virtual NewModels::Base::StringPairVec getExtraGlobalParams() const{ return {}; }
+};
+
+//----------------------------------------------------------------------------
+// CurrentSourceModels::ACSource
+//----------------------------------------------------------------------------
+//! AC source
+/*! It has 6 parameters:
+
+    - \c tStart - time at which the current starts [ms]
+    - \c tStop  - time at which the current stops [ms]
+    - \c amp    - amplitude of the sine wave [nA]
+    - \c freq   - frequency of the sine wave [Hz]
+    - \c phase  - phase of the sine wave [rad]
+    - \c offset - offset of the sine wave [nA]
+*/
+class ACSource : public Base
```

Comment:
I think this model is too PyNN-specific to be included in GeNN - I've never really seen it used

Reply:
and the derived parameter `omega` in it also produces warnings due to unused argument `dt` - another reason to drop it :laughing:

Reply:
Exactly :smile: 

(if you remove the name of the parameter from the lambda function (or any other function) declaration i.e. ``[](const vector<double> &pars, double){...}`` the warning goes away btw)